ydb/src/client_topic/topicwriter/writer.rs

```diff
@@ -0,0 +1,405 @@
+                    break;
+                }
+            }
+        }
+        if !messages.is_empty() {
+            println!("Message is ACTUALLY sent");
+            task_params
+                .request_stream
+                .send(stream_write_message::FromClient {
+                    client_message: Some(ClientMessage::WriteRequest(WriteRequest {
+                        messages,
+                        codec: 1,
+                    })),
+                })
+                .unwrap(); // TODO: HANDLE ERROR
+        }
+        Ok(())
+    }
+
+    async fn receive_messages_loop_iteration(
+        server_messages_receiver: &mut AsyncGrpcStreamWrapper<
+            stream_write_message::FromClient,
+            stream_write_message::FromServer,
+        >,
+        confirmation_reception_queue: &mut Arc<Mutex<TopicWriterReceptionQueue>>,
+    ) -> YdbResult<()> {
+        match server_messages_receiver.receive().await {
+            Ok(server_message) => match server_message {
+                ServerMessage::InitResponse(init_response_body) => {
+                    return Err(YdbError::Custom(
+                        "Unexpected message type in stream reader: init_response".to_string(),
+                    ));
+                }
+                ServerMessage::WriteResponse(write_response_body) => {
+                    for raw_ack in write_response_body.acks {
+                        let write_ack = WriteAck::from(raw_ack);
+                        let mut reception_queue = confirmation_reception_queue.lock().unwrap();
+                        let reception_ticket = reception_queue.try_get_ticket();
+                        match reception_ticket {
+                            None => {
+                                return Err(YdbError::Custom(
+                                    "Expected reception ticket to be actually present".to_string(),
+                                ));
+                            }
+                            Some(mut ticket) => {
+                                if write_ack.seq_no != ticket.get_seq_no() {
+                                    return Err(YdbError::Custom(
+                                        "Reception ticket and write ack seq_no mismatch"
+                                            .to_string(),
+                                    ));
+                                }
+                                ticket.send_confirmation_if_needed(write_ack.status);
+                            }
+                        }
+                    }
+                }
+                ServerMessage::UpdateTokenResponse(_update_token_response_body) => {}
+            },
+            Err(some_err) => {
+                return Err(YdbError::from(some_err));
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn write(&mut self, mut message: TopicWriterMessage) -> YdbResult<()> {
+        self.is_cancelled().await?;
+
+        self.last_seq_num_handled += 1;
```

Comment:
if self.auto_set_seq_no is false - last_seq_num_handled not store real last seq_num, because customer messages can set seqno with holes. For example:
1, 2, 3, 100 - it is OK.

Need rename the field or store real last seq